sdk/identity/azure-identity/src/token_cache.cpp

```diff
@@ -120,3 +117,44 @@ AccessToken TokenCache::GetToken(
   item->AccessToken = newToken;
   return newToken;
 }
+
+namespace {
+
+// Compile-time Fibonacci sequence computation.
+// Get() produces a std::array<T> containing the numbers in ascending order.
+template <
+    typename T, // Type
+    T L = 0, // Left hand side
+    T R = 1, // Right hand side
+    size_t N = 0, // Counter (for array)
+    bool X = ((std::numeric_limits<T>::max() - L) < R)> // Condition to stop (integer overflow of T)
+struct SortedFibonacciSequence
+{
+  static constexpr auto Get();
+};
+
+template <typename T, T L, T R, size_t N> struct SortedFibonacciSequence<T, L, R, N, true>
+{
+  static constexpr auto Get()
+  {
+    std::array<T, N + 1> result{};
+    result[N] = L;
+    return result;
+  }
+};
+
+template <typename T, T L, T R, size_t N, bool X>
+constexpr auto SortedFibonacciSequence<T, L, R, N, X>::Get()
+{
+  auto result = SortedFibonacciSequence<T, R, R + L, N + 1>::Get();
+  result[N] = L;
+  return result;
+}
+
+template <typename T> bool ShouldCleanUpCacheFromExpiredItems(T cacheSize)
+{
+  static auto const Fibonacci = SortedFibonacciSequence<T, 1, 2>::Get();
```

Comment:
Why don't we need to specify a `size_T N` counter? What is that template parameter meant for?

Reply:
It is needed for array - so that when template "stops", we know the size of std::array that we need to create. And on each step we use N to assign the current sum to the corresponding index in the resulting array.